### ISSUE TITLE
jax.lax.exp2 gradient overflows to inf where finite float64 derivative is expected

### ISSUE THREAD

**Issue by ALinrunrun** [jax-ml/jax#40382]
### Description

`jax.lax.exp2` appears to return an incorrect non-finite first-order derivative for a finite `float64` input.

For:

```
f(x) = 2**x
```

the derivative is:

```
f'(x) = ln(2) * 2**x
```

At `x = 1024.0000000000002`, the high-precision reference derivative rounded to `float64` is approximately `1.2460659279419801e+308`, which is finite and normal. However, JAX returns `inf`.

The same issue is also relevant to the mapped public path `jax.numpy.exp2`.

### Minimal reproducible example

```
#!/usr/bin/env python3
import os
import sys
import math

os.environ.setdefault("CUDA_VISIBLE_DEVICES", "-1")
os.environ.setdefault("JAX_PLATFORMS", "cpu")
os.environ.setdefault("TF_CPP_MIN_LOG_LEVEL", "3")

try:
    import jax
    import jax.numpy as jnp
except ImportError as e:
    print(f"missing dep: {e}")
    sys.exit(2)

jax.config.update("jax_enable_x64", True)

x = 1024.0000000000002
expected = 1.2460659279419801e+308

fn = jax.lax.exp2
actual = float(jax.grad(lambda z: jnp.sum(fn(z)))(jnp.asarray(x, dtype=jnp.float64)))

print("jax:", actual)
print("expected:", expected)

if not (math.isfinite(actual) and abs(actual - expected) <= 1e-6 * abs(expected)):
    sys.exit(0)
sys.exit(1)
```

### Actual result

```
jax: inf
expected: 1.2460659279419801e+308
```

### Expected result

```
1.2460659279419801e+308
```

The expected derivative is finite, normal, and representable in `float64`. JAX instead returns `inf`, suggesting that the autodiff rule or its internal computation overflows before producing the final derivative.

### System info (python version, jaxlib version, accelerator, etc.)

jax: 0.10.1
jaxlib: 0.10.1
numpy: 2.4.6
python: 3.13.13
platform: Ubuntu 24.04.3 LTS, Linux x86_64
accelerator: CPU backend

Note: GPU is present, but CUDA-enabled jaxlib is not installed, so JAX falls back to CPU.

**Comment by jakevdp** [comment]
This falls under #38651 – issues like this (typically discovered via AI fuzzing of extreme input values) are low priority for the JAX team.
